### PR TITLE
MNT: refactor _make_stc

### PR DESCRIPTION
**Pull request by massich**
This PR refactors _make_stc in order to simplify #6609

### CODE REVIEW

**Comment by massich** [comment]
For now just runnint the CIs.

**Comment by massich** [comment]
I thought I would also win something in terms of speed with the vectorization but that's not the case. `test_make_inverse_operator_vector` takes the same time with both implementations.

**Comment by massich** [comment]
Travis errors are unrelated:
```sh
>               raise URLError(err)
3391E               urllib.error.URLError: <urlopen error timed out>
```

File: mne/source_estimate.py [hunk 1]
```diff
@@ -378,53 +378,45 @@ def _get_src_type(src, vertices, warn_text=None):
...
-        src_type = _get_src_type(src=None, vertices=vertices,
-                                 warn_text=warn_text)
+    def guess_src_type():
+        return _get_src_type(src=None, vertices=vertices, warn_text=warn_text)
 
+    src_type = guess_src_type() if src_type is None else src_type
+
+    if vector and src_type == 'mixed':  # XXX this should be supported someday
+        raise NotImplementedError(
+            'Vector source estimates for mixed source spaces is not supported'
```
Comment (wmvanvliet) [inline mne/source_estimate.py]: *are* not supported

**Comment by wmvanvliet** [comment]
Great refactor! Looks much cleaner. Since you touched the rotation logic, did you double check that the vectors in the VectorSourceEstimate are rotated correctly?

**Comment by massich** [comment]
@wmvanvliet yes. If you check the first commit bd2b703, I ran both computations, put assert and run the entire test suite. All green lights!

```py
            xx = np.matmul(
                np.transpose(source_nn.reshape(n_vertices, 3, 3),
                             axes=[0, 2, 1]),
                data.reshape(n_vertices, 3, -1)
            )
            np.testing.assert_allclose(xx, data_rot)
```

File: mne/source_estimate.py (hunk 1 above)
Comment (massich) [reply mne/source_estimate.py]: fixed in 2f6c1b04f

**Pull request merged**